lxd/device/cdi/spec.go

```diff
@@ -0,0 +1,104 @@
+package cdi
+
+import (
+	"fmt"
+	"os/exec"
+	"path/filepath"
+
+	"github.com/NVIDIA/nvidia-container-toolkit/pkg/nvcdi"
+	cdiSpec "tags.cncf.io/container-device-interface/pkg/cdi"
+	"tags.cncf.io/container-device-interface/specs-go"
+
+	"github.com/canonical/lxd/shared"
+	"github.com/canonical/lxd/shared/logger"
+)
+
+const (
+	// defaultNvidiaTegraMountSpecPath is default location of CSV files that define the modifications required to the OCI spec.
+	defaultNvidiaTegraMountSpecPath = "/etc/nvidia-container-runtime/host-files-for-container.d"
+)
+
+// DefaultNvidiaTegraCSVFiles returns the default CSV files for the Nvidia Tegra platform.
+func DefaultNvidiaTegraCSVFiles(rootPath string) []string {
+	files := []string{
+		"devices.csv",
+		"drivers.csv",
+		"l4t.csv",
+	}
+
+	var paths []string
```

Comment:
This should be pre-allocated to have the capacity of len(files)

Reply:
fixed